ui/components/multichain/network-list-menu/network-list-menu.js

```diff
@@ -80,14 +89,51 @@ export const NetworkListMenu = ({ onClose }) => {
 
   const showSearch = nonTestNetworks.length > 3;
 
+  const orderedNetworksList = useSelector(getOrderedNetworksList);
+
+  const newOrderNetworks = () => {
+    if (orderedNetworksList.length === 0) {
+      return nonTestNetworks;
+    }
+
+    // Reorder nonTestNetworks based on the order of chainIds in orderedNetworksList
+    const sortedNetworkList = orderedNetworksList
+      .map((chainId) =>
+        nonTestNetworks.find((network) => network.chainId === chainId),
+      )
+      .filter(Boolean);
+
+    return sortedNetworkList;
+  };
+
+  const networksList = newOrderNetworks();
+
   useEffect(() => {
     if (currentlyOnTestNetwork) {
       dispatch(setShowTestNetworks(currentlyOnTestNetwork));
     }
   }, [dispatch, currentlyOnTestNetwork]);
   const [searchQuery, setSearchQuery] = useState('');
+  const [items, setItems] = useState([...networksList]);
+  const [showBanner, setShowBanner] = useState(true);
+
+  const onBannerClose = () => {
+    setShowBanner(false);
+  };
 
-  let searchResults = [...nonTestNetworks];
+  const onDragEnd = (result) => {
```

Comment:
Can some of this be moved to an action so we aren't doing data management here?

Reply:
I think we should keep it here as it's specific to network list drag and drop